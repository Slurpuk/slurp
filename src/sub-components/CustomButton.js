import React from 'react';
import {
  Text,
  View,
  StyleSheet,
  Pressable,
  Animated,
  Image,
  TouchableOpacity,
  Dimensions,
} from 'react-native';

export default function CustomButton(props) {
  const {priority, text, optionalNumber = null} = props;
  return (
    <View>
      <Animated.View style={{transform: [{scale}]}}>
        <TouchableOpacity
          style={[buttonStyles.outer, buttonStyles[priority]]}
          activeOpacity={1}
          onPressIn={onPressIn}
          onPressOut={onPressOut}
          onPress={props.onPress}
        >
          <Text style={buttonStyles.buttonText}>{text}</Text>
          {optionalNumber == null ? (
            <></>
          ) : (
            <Text
              style={[
                buttonStyles.optionalNumber,
                buttonStyles[`bubble${priority}`],
              ]}
            >
              {optionalNumber}
            </Text>
          )}
        </TouchableOpacity>
      </Animated.View>
    </View>
  );
}
const animation = new Animated.Value(0);
const inputRange = [0, 1];
const outputRange = [1, 0.8];
const scale = animation.interpolate({inputRange, outputRange});

const onPressIn = () => {
  Animated.spring(animation, {
    toValue: 0.095,
    speed: 100,
    useNativeDriver: true,
  }).start();
};
const onPressOut = () => {
  Animated.spring(animation, {
    toValue: 0,
    speed: 70,
    useNativeDriver: true,
  }).start();
};
const screenWidth = Dimensions.get('window').width;

const buttonStyles = StyleSheet.create({
  primary: {
    backgroundColor: '#087562',
  },

  secondary: {
    backgroundColor: '#2D466B',
  },

  optionalNumber: {
    borderRadius: 30,
    width: 20,
    height: 20,
    textAlign: 'center',
    marginBottom: 15,
    marginLeft: 8,
  },

  bubbleprimary: {
    backgroundColor: '#183342',
  },

  bubblesecondary: {
    backgroundColor: 'whitesmoke',
    color: '#555555',
  },

  buttonText: {
    color: '#EFEFEF',
    fontFamily: 'Poppins-SemiBold',
    letterSpacing: 0.5,
    fontSize: 17,
  },

  outer: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 13,
    marginHorizontal: 10,
    height: 43,
    width: screenWidth * 0.91,
    // borderWidth: 2,
  },
});
