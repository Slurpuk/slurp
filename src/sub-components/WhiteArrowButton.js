import React, {useEffect} from 'react';
import {BackHandler, Pressable} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

const WhiteArrowButton = ({direction = 'back', navigation}) => {
  let iconName;
  switch (direction) {
    case 'left':
      iconName = 'md-chevron-back-circle';
      break;
    case 'right':
      iconName = 'md-chevron-forward-circle';
      break;
    case 'up':
      iconName = 'md-chevron-up-circle';
      break;
    case 'down':
      iconName = 'md-chevron-down-circle';
      break;
  }

  // function handleBackButtonClick() {
  //   navigation.navigate('Home');
  //   return true;
  // }
  //
  // useEffect(() => {
  //   BackHandler.addEventListener('hardwareBackPress', handleBackButtonClick);
  //   return () => {
  //     BackHandler.removeEventListener(
  //       'hardwareBackPress',
  //       handleBackButtonClick,
  //     );
  //   };
  // }, []);

  return (
    <Pressable
      underlayColor={'transparent'}
      onPress={() => navigation.goBack()}
      style={({pressed}) => [
        {
          underlayColor: 'gray',
          opacity: pressed ? 0.6 : 1,
        },
      ]}>
      <Icon name={iconName} color={'white'} size={34} />
    </Pressable>
  );
};

export default WhiteArrowButton;
