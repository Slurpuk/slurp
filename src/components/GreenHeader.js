import React from 'react';
import {
  StyleSheet,
  View,
  Pressable,
  Image,
  Text,
  Alert,
  TouchableHighlight,
  StatusBar,
  Platform,
} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import {getStatusBarHeight} from 'react-native-status-bar-height';
import WhiteArrowButton from '../SubComponents/WhiteArrowButton';

const GreenHeader = ({headerText}) => {
  const onBackButtonClicked = () => {
    Alert.alert('FUTURE NAVIGATION FEATURE', 'Go back to previous page', [
      {
        text: 'OK',
      },
    ]);
  };

  return (
    <View style={styles.header}>
      <StatusBar translucent={true} backgroundColor="transparent" />
      <View style={styles.arrow}>
        <WhiteArrowButton direction={'left'} onPress={onBackButtonClicked} />
      </View>
      <Text style={styles.headerText}>{headerText}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    backgroundColor: '#046D66',
    display: 'flex',
    paddingTop: getStatusBarHeight(),
    paddingBottom: '6%',
    paddingHorizontal: '5%',
    alignItems: 'center',
  },

  headerText: {
    color: '#EDEBE7',
    fontWeight: '700',
    fontStyle: 'normal',
    fontSize: 27,
    marginLeft: 20,
    letterSpacing: 0.4,
},
});

export default GreenHeader;
