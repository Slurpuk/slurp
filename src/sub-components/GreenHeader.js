import React from 'react';
import {StyleSheet, View, Text, Alert, StatusBar, Platform} from 'react-native';
import {getStatusBarHeight} from 'react-native-status-bar-height';
import WhiteArrowButton from './WhiteArrowButton';
import textStyles from '../../stylesheets/textStyles';
import {getTightPaddingTop} from '../../stylesheets/StyleFunction';

const GreenHeader = ({headerText, navigation}) => {
  return (
    <View style={styles.header}>
      <StatusBar translucent={true} backgroundColor="transparent" />
      <View style={styles.arrow}>
        <WhiteArrowButton direction={'left'} navigation={navigation} />
      </View>
      <View style={styles.title}>
        <Text style={textStyles.whiteTextOnGreenHeaderText}>{headerText}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    backgroundColor: '#046D66',
    paddingTop: getTightPaddingTop(),
    paddingBottom: '6%',
    paddingHorizontal: '5%',
    alignItems: 'center',
  },
  title: {
    marginTop: '2%',
  },
});

export default GreenHeader;
