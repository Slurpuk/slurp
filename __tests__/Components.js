import renderer from 'react-test-renderer';
import React from 'react';
import CustomButton from '../src/sub-components/CustomButton';
import GreenHeader from '../src/sub-components/GreenHeader';
import WhiteArrowButton from '../src/sub-components/WhiteArrowButton';
import Icon from 'react-native-vector-icons/Ionicons';
import {Alert} from 'react-native';
import {render} from '@testing-library/react';

describe('Components', function () {
  const spyAlert = jest
    .spyOn(Alert, 'alert')
    //@ts-ignore
    .mockImplementation((title, message, callbackOrButtons) =>
      callbackOrButtons[1].onPress(),
    );

  afterEach(() => spyAlert.mockClear());

  describe('Custom sub-components', function () {
    describe('button', function () {
      it('should render correctly', function () {
        const tree = renderer.create(<CustomButton />).toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should render correctly with number indicator', function () {
        const tree = renderer
          .create(<CustomButton optionalNumber={5} />)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('white arrow button', function () {
    jest.mock('react-native-vector-icons/Ionicons', () => 'Icon');

    it('should render correctly if the arrow points back', function () {
      const tree = renderer
        .create(<WhiteArrowButton direction={'left'} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly if the arrow points right', function () {
      const tree = renderer
        .create(<WhiteArrowButton direction={'right'} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly if the arrow points up', function () {
      const tree = renderer
        .create(<WhiteArrowButton direction={'up'} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly if the arrow points down', function () {
      const tree = renderer
        .create(<WhiteArrowButton direction={'down'} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
