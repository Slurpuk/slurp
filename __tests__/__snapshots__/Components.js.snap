// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom components button should render correctly 1`] = `
<View>
  <View
    collapsable={false}
    style={
      Object {
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 13,
          "display": "flex",
          "flexDirection": "row",
          "height": 43,
          "justifyContent": "center",
          "opacity": 1,
          "width": 682.5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#EFEFEF",
            "fontFamily": "Poppins-SemiBold",
            "fontSize": 17,
            "letterSpacing": 0.5,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Custom components: button should render correctly 1`] = `
<View>
  <View
    collapsable={false}
    style={
      Object {
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 13,
          "display": "flex",
          "flexDirection": "row",
          "height": 43,
          "justifyContent": "center",
          "opacity": 1,
          "width": 682.5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#EFEFEF",
            "fontFamily": "Poppins-SemiBold",
            "fontSize": 17,
            "letterSpacing": 0.5,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Custom sub-components button should render correctly 1`] = `
<View>
  <View
    collapsable={false}
    style={
      Object {
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 13,
          "display": "flex",
          "flexDirection": "row",
          "height": 43,
          "justifyContent": "center",
          "opacity": 1,
          "width": 682.5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#EFEFEF",
            "fontFamily": "Poppins-SemiBold",
            "fontSize": 17,
            "letterSpacing": 0.5,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Custom sub-components green header should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#046D66",
      "flexDirection": "row",
      "paddingBottom": "6%",
      "paddingHorizontal": "5%",
      "paddingTop": 30,
    }
  }
>
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "opacity": 1,
            "underlayColor": "gray",
          },
        ]
      }
      underlayColor="transparent"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 34,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ïˆ‰
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": "2%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#EDEBE7",
          "fontFamily": "JosefinSans-Bold",
          "fontSize": 27,
          "fontStyle": "normal",
          "includeFontPadding": false,
          "letterSpacing": 0.4,
          "marginLeft": 20,
        }
      }
    />
  </View>
</View>
`;
